### PR TITLE
Fix sample doc pages not validating

### PR DESCRIPTION
**Pull request by sebastianbenz**
The samples define head tags are injected into the template
causing validation errors due to duplicated tags. I couldn't
come up with a clean handle this on a template level. Instead,
I've opted for writing a optimizer transformer that'll remove
duplicates.

### CODE REVIEW

File: platform/lib/utils/HeadDedupTransformer.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    name: 'viewport',
+  },
+  link: {
+    rel: 'canonical',
+  },
+};
+
+/**
+ * A custom transformer for the @ampproject/toolbox-optimizer rewriting
+ * CSS selectors to save bytes in <style amp-custom>.
```
Comment (tharders) [inline platform/lib/utils/HeadDedupTransformer.js]: This text does not fit.

File: platform/lib/utils/HeadDedupTransformer.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/**
```
Comment (tharders) [inline platform/lib/utils/HeadDedupTransformer.js]: The filename should start with a lowercase character

Comment (sebastianbenz) [reply platform/lib/utils/HeadDedupTransformer.js]: uppercase when exporting a class

**Comment by sebastianbenz** [comment]
Some amp components require custom head elements, e.g. amp-access configures it's auth endpoints.

File: platform/lib/utils/HeadDedupTransformer.js (hunk 1 above)
Comment (sebastianbenz) [reply platform/lib/utils/HeadDedupTransformer.js]: done

**Comment by sebastianbenz** [comment]
PTAL

**Comment by sebastianbenz** [comment]
👍

**Pull request merged**